pkg/controller/account/ec2.go

```diff
@@ -58,14 +81,56 @@ func (r *ReconcileAccount) InitializeRegion(reqLogger logr.Logger, account *awsv
 		AwsRegion:               region,
 	})
 	if err != nil {
-		connErr := fmt.Sprintf("Unable to connect to region: %s when attempting to initialize it", region)
+		connErr := fmt.Sprintf("unable to connect to region %s when attempting to initialize it", region)
 		reqLogger.Error(err, connErr)
 		// Notify Error channel that this region has errored and to move on
 		ec2Errors <- connErr
 
 		return err
 	}
 
+	// If the quota is 0, there was an error and we cannot act on it
+	if vCPUQuota != 0 {
+		// Check if a request is necessary
+		// If there are errors, this will return false, and will not continue to try
+		// to set the quota
+		quotaIncreaseRequired, err = getVCPUQuota(awsClient, vCPUQuota)
+		if err != nil {
+			reqLogger.Error(err, "failed retriving current vCPU quota from AWS")
+		}
+	}
+
+	if quotaIncreaseRequired {
+		reqLogger.Info("vCPU quota increase required", "region", region)
+		caseID, err = checkQuotaRequestHistory(awsClient, vCPUQuota)
+		if err != nil {
+			reqLogger.Error(err, "failed retriving quota change history")
```

Comment:
Is there any error here that we want to check and possibly return on?

Reply:
I've tried to handle all the error that could reasonably be handled within that function using the retry and conditional exceptions.

With regard to returning, I think it's more appropriate to log a quota request failure, but allow the initialization to continue, as not having the default quota is not necessarily a killer, and I was concerned about bombing out the entire goroutine.

I suppose with Eric's recent changes to add the initializing state, that might not be critical.  I'm happy to take suggestions on whether we should do so or not.

Reply:
I think its fine as it if you don't see any obvious error to return on. Just want to ensure that all paths were considered since the most that ever happens with this specific err is printing it.

Reply:
small brain dump (apply that adjective however you like...)

It doesn't make sense for a thing that's going to be run in a goroutine to return an error (or anything at all). So that just needs to be removed from the prototype and flows. That's something I left behind when I did #454 because prior to that the error conditions were being ignored anyway.

Errors in the goroutine can only be dealt with via logging and CR status. If we deem an error condition to be fatal, set the CR to `Failed`. (Optimization: cancel any other sub-goroutines.)